WordPress/gutenberg
# Add support for button elements to theme.json

## What?
Adds support for button elements to theme.json. This means themes can now add styles to their buttons using theme.json like this:

```
{
	"styles": {
		"elements": {
			"button": {
				"color": {
					"background": "red"
				}
			}
		}
	}
}
```

## Why?
We need a consistent way to style buttons across blocks. The search block is the most obvious example here, but there will also be third party blocks that will benefit from this change.

## How?
Adds a new class, `wp-element-button` to buttons, so that they can share the same styles. The search block requires a server side implementation. Right now we don't share the class name between client and server.

## Testing Instructions
- Check out this PR: https://github.com/Automattic/themes/pull/5849
- Switch to Archeo
- Confirm that in both the editor and the front end the buttons in the button block and search blocks both look the same.

## Screenshots or screencast 
Frontend:
<img width="710" alt="Screenshot 2022-04-12 at 13 42 36" src="https://user-images.githubusercontent.com/275961/162964921-798b88dd-13f8-436b-90a5-60cc93088d74.png">

Editor:
<img width="606" alt="Screenshot 2022-04-12 at 13 42 42" src="https://user-images.githubusercontent.com/275961/162964906-c50a132d-6adc-41f6-93b8-824c0c682507.png">



cc @WordPress/block-themers
